python/social_bot/tasks.py

```diff
@@ -196,10 +220,26 @@ def _push_reward_queue(self, value):
         if (value > 0 and len(self._q) == self._max_reward_q_length
                 and sum(self._q) >= self._max_reward_q_length *
                 self._reward_thresh_to_increase_range):
-            self._random_range *= 1. + self._increase_range_by_percent
-            if self._random_range > self._orig_random_range:
-                self._random_range = self._orig_random_range
-            logging.info("Raising random_range to %f", self._random_range)
+            if self._also_curriculum_target_angle:
+                self._random_angle += 20
+                logging.info("Raising random_angle to %d", self._random_angle)
+            if (not self._also_curriculum_target_angle or
+                self._random_angle > 360):
+                self._random_angle = 60
+                new_range = min(
+                    (1. + self._increase_range_by_percent
+                        ) * self._random_range,
+                    self._orig_random_range)
+                new_control_range = min(1.0,
+                    (1. + self._increase_range_by_percent) *
```

Comment:
Any experiments on increasing control_range? From my experiences, at the start of training progress actions are more random, so small ranges make the exploration not efficient. (Pure random actions have a (v*t)^(1/2) expectation explore range, v is the average speed).  A higher control range should be at start.  So I've already set a relatively larger initial action range in the "effort_limit" term of joints in model file. There is also an optional action_cost argument in playground to encourage the agent using limited actions as possible while completing its main task. As it is already higher enough in the beginning, In my view increasing control_range might not be so meaningful.

Also, as the magnitude of actions changes in (s, a, s', r), it might have unknown affect on off-policy RL algorithm. MDP become POMDP, I'm not quite sure but theoretically we may need to add new_control_range to observation.

Reply:
I think target random_range is fine because the agent have the target pos in observation. It's just a new range of an existing feature in observation. If random_range changes, target position in the next state s' still depends on last states and actions, does not depends on random_range. 

But if action_range changes, it's a different case. The next state s' depends on last states, actions and action_range.

For the flipping problem, either we can set the effort_limit in model file a little bit lower, or add a slightly action_cost term. 

Another way is to set the joint using a PID controller rather than force controlling directly. It's vey simple, just by editing the agent config file in "agent.cfg" to enable this. I would suggest this way and I already set Youbot and iCub joints to PID. The action is the target speed of PID controller input. Actually most real robots, robot arms or quadcopters using sub-level PID controller, either velocity or position way.

Force controlling is dangers because F/m is the acceleration, given enough time*action for one side of wheels the velocity could be very high, even action range is smaller.